src/main/webapp/modal.js

```diff
@@ -1,8 +1,54 @@
+// Used to decide, if multiple modals are open simultanesouly, which one should receive
+// tab commands for accessibility.  Because relatively few modals should be open at any
+// given time (< 10), we can use a simple list
+const visibleModals = [];
+const focusableElements = 'button, input, select, textarea';
+let firstElement = null;
+let lastElement = null;
+
+
+function updateFocusableElements() {
+  if (visibleModals.length <= 0) {
+    firstElement = null;
+    lastElement = null;
+    return;
+  }
+
+  let modalToFocus = visibleModals[visibleModals.length - 1];
+  let focusableElements = modalToFocus.container.querySelectorAll('button, input, select, textarea');
+
+  firstElement = focusableElements[0];
+  lastElement = focusableElements[focusableElements.length - 1];
+}
+
+document.addEventListener('keydown', function(e) {
+  if (visibleModals.length <= 0 || firstElement == null || lastElement == null) {
+    return;
+  }
+
+  let isTabPressed = e.keyCode == 9;
```

Comment:
can we make `9` a constant, or maybe add a comment? I am not sure which key is 9